qa/full-cluster-restart/src/javaRestTest/java/org/elasticsearch/upgrades/ParameterizedFullClusterRestartTestCase.java

```diff
@@ -39,6 +42,9 @@ public abstract class ParameterizedFullClusterRestartTestCase extends ESRestTest
     private static IndexVersion oldIndexVersion;
     private static boolean upgradeFailed = false;
     private static boolean upgraded = false;
+
+    private static boolean oldClusterFeaturesInitialized = false;
+    private static final Set<String> oldClusterFeatures = new HashSet<>();
```

Comment:
Rather than a separate boolean indicating if its initialized, couldnt you have `oldClusterFeatures == null` indicating uninitialized?

Reply:
Yeah, I don't know what I like the least: null check or separate variables who are semantically linked together :)
My preference would be using functions, but it's a different style.
Probably the null check is the least evil, especially since it will be behind an assertion. I will change that.